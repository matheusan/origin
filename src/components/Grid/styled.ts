import styled from 'styled-components';

const columnSizes = {
  1: '100%',
  2: '48%',
  3: '32%',
  4: '23.7%',
  5: '18.7%'
};

export const Wrapper: JSX.Element = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: ${props => (props.nowrap ? 'nowrap' : 'wrap')};
  align-items: center;
  align-content: stretch;

  padding: 0;
`;

export const Column: JSX.Element = styled.div`
  margin-right: 16px;
  ${props =>
    props.columns
      ? `width: ${columnSizes[props.columns]};`
      : 'flex: 1 1 200px;'}

  &:last-child {
    margin-right: 0;
  }

  @media only screen and (max-width: 380px) {
    flex: auto;
    margin-right: 0;
    margin-bottom: 16px;
    width: 100%;
  }
`;
