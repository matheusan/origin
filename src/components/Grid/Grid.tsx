import * as React from 'react';

import * as S from './styled';

export interface GridProps {
  nowrap?: boolean;
}

const Grid: React.SFC<GridProps> = ({ nowrap = false, children }) => (
  <S.Wrapper nowrap={nowrap}>
    {[].concat(children).map((child, i) => (
      <S.Column key={i}>{child}</S.Column>
    ))}
  </S.Wrapper>
);

export default Grid;
