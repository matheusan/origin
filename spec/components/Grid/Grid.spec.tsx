import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Grid from '../../../src/components/Grid';

describe('Grid', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Grid>
        <span>One thing</span>
        <span>Two things</span>
      </Grid>
    );
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one element', () => {
    wrapper = shallow(
      <Grid>
        <span>One thing</span>
      </Grid>
    );
    expect(wrapper.find('span').length).toBe(1);
  });

  it('should render wathever child element is given', () => {
    wrapper = shallow(
      <Grid>
        <span>One thing</span>
        <button>My button</button>
        <img src="noop.jpg" />
      </Grid>
    );
    expect(wrapper.find('span').length).toBe(1);
    expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('img').length).toBe(1);
  });
});
