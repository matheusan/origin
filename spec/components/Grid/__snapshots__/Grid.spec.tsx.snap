// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders 1`] = `
<styled.div
  nowrap={false}
>
  <styled.div
    key="0"
  >
    <span>
      One thing
    </span>
  </styled.div>
  <styled.div
    key="1"
  >
    <span>
      Two things
    </span>
  </styled.div>
</styled.div>
`;
